src/tests/class_tests/openms/source/IsoSpec_test.cpp

```diff
@@ -282,24 +282,17 @@ END_SECTION
 START_SECTION(( bool IsoSpecTotalProbGeneratorWrapper::nextConf() ))
 {
   double total_prob = 0.99999;
-  bool do_trim = true;
 
-  IsoSpecTotalProbGeneratorWrapper ITPW(EmpiricalFormula("C6H12O6"), total_prob, do_trim);
+  IsoSpecTotalProbGeneratorWrapper ITPW(EmpiricalFormula("C6H12O6"), total_prob);
   TEST_EQUAL(compare_generator_to_reference(ITPW, fructose_expected_oms, -1), true);
 
-  IsoSpecTotalProbGeneratorWrapper ITPW2(EmpiricalFormula("C6H12O6"), total_prob, do_trim);
-  TEST_EQUAL(generator_length(ITPW2), 17);
+  IsoSpecTotalProbGeneratorWrapper ITPW2(EmpiricalFormula("C6H12O6"), total_prob);
+  TEST_EQUAL(generator_length(ITPW2), 2548); // Total prob is a hint, if we keep extracting configurations
```

Comment:
because there is not trimming?

Reply:
Yes. The way it used to be is that TotalProbGenerator used to be kind of a hybrid between a generator and something that stored configurations and performed p-trimming. This led to it being somewhat memory inefficient, and unclear in design. Now the two parts are separated and the generator just goes over the entire configuration space in an order that's close enough to sorted to be able to perform the p-trim efficiently (while not exactly sorted to avoid the nlogn complexity), and another piece of code accumulates the configurations, terminates the search in the appropriate moment and performs the p-trim.